File: SWP391-SE1633NET-G2-main/LearningProject/web/assets/js/add_lesson.js
/* 
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/JSP_Servlet/JavaScript.js to edit this template
 */

async function doSubmit() {
    // get video id from video-link
    try {
        const videoLink = document.querySelector('input[name="url"]').value;
        const videoId = videoLink.split('watch?v=')[1];

        const API_KEY = "AIzaSyBBcUOX-BytFmB96iWqMBIsRLzTgP0hhWQ";

        // use google api for get video detail
        const duration = await fetch(`https://www.googleapis.com/youtube/v3/videos?id=${videoId}&part=contentDetails&key=${API_KEY}`).then(res => res.json()).then(res => res.items[0].contentDetails.duration);
        console.log(duration);
        const parseDuration = (d) => {
            const match = d.match(/PT(\d+H)?(\d+M)?(\d+S)?/);
            const hours = (parseInt(match[1]) || 0);
            const minutes = (parseInt(match[2]) || 0);
            const seconds = (parseInt(match[3]) || 0);
            return hours * 3600 + minutes * 60 + seconds;
        };

        // parse duration to seconds
        const durationInSeconds = parseDuration(duration);

        // set video-length value
        document.querySelector('input[name="duration"]').value = durationInSeconds * 1000;
        document.querySelector('form[data-type="video"]').submit();
        //document.querySelector('form').submit();
    } catch (error) {
        alert('Invalid video link');
    }
}

function Forward(target) {
    if (target.getAttribute("data-lesson-type") == "Video") {
        document.getElementById("LessonVideoName").value = target.getAttribute("data-lesson-name");
        document.getElementById("VideoLink").value = target.getAttribute("data-video-url");
        document.getElementById("LessonVideoId").value = target.getAttribute("data-lesson-id");
    } else if (target.getAttribute("data-lesson-type") == "Docs") {
        tinyMCE.activeEditor.setContent('');
        document.getElementById("LessonDocsName").value = target.getAttribute("data-lesson-name");
        document.getElementById("Time").value = target.getAttribute("data-lesson-time");        
        document.getElementById("LessonDocsId").value = target.getAttribute("data-lesson-id");
        tinymce.activeEditor.selection.setContent(document.getElementById("docs-content-" + document.getElementById("LessonDocsId").value).value);
    } else {
        document.getElementById("LessonQuizTitle").value = target.getAttribute("data-lesson-name");
        document.getElementById("LessonQuizTime").value = target.getAttribute("data-lesson-time");
        document.getElementById("LessonQuizId").value = target.getAttribute("data-lesson-id");
    }
}

//function doSubmit2() {
//    // get video id from video-link
//
//    const videoLink = document.querySelector('input[name="VideoLink"]').value;
//    const videoId = videoLink.split('watch?v=')[1];
//
//    const API_KEY = "AIzaSyBBcUOX-BytFmB96iWqMBIsRLzTgP0hhWQ"
//
//    // use google api for get video detail
//    fetch(`https://www.googleapis.com/youtube/v3/videos?id=${"${videoId}"}&part=contentDetails&key=${"${API_KEY}"}`)
//            .then(res => res.json())
//            .then(res => res.items[0].contentDetails.duration).then(duration => {
//        console.log(duration);
//        const parseDuration = (d) => {
//            const match = d.match(/PT(\d+H)?(\d+M)?(\d+S)?/);
//            const hours = (parseInt(match[1]) || 0);
//            const minutes = (parseInt(match[2]) || 0);
//            const seconds = (parseInt(match[3]) || 0);
//            return hours * 3600 + minutes * 60 + seconds;
//        };
//
//        // parse duration to seconds
//        const durationInSeconds = parseDuration(duration);
//
//        // set video-length value
//        document.querySelector('input[name="time_duration"]').value = durationInSeconds * 1000;
//        document.querySelector('form[data-type="Video"]').submit();
//    }).catch(error => alert('Invalid video link'));
//
//}

async function doSubmit2() {
    // get video id from video-link
    try {
        const videoLink = document.querySelector('input[name="VideoLink"]').value;
        const videoId = videoLink.split('watch?v=')[1];

        const API_KEY = "AIzaSyBBcUOX-BytFmB96iWqMBIsRLzTgP0hhWQ";

        // use google api for get video detail
        const duration = await fetch(`https://www.googleapis.com/youtube/v3/videos?id=${videoId}&part=contentDetails&key=${API_KEY}`).then(res => res.json()).then(res => res.items[0].contentDetails.duration);
        console.log(duration);
        const parseDuration = (d) => {
            const match = d.match(/PT(\d+H)?(\d+M)?(\d+S)?/);
            const hours = (parseInt(match[1]) || 0);
            const minutes = (parseInt(match[2]) || 0);
            const seconds = (parseInt(match[3]) || 0);
            return hours * 3600 + minutes * 60 + seconds;
        };

        // parse duration to seconds
        const durationInSeconds = parseDuration(duration);

        // set video-length value
        document.querySelector('input[name="time_duration"]').value = durationInSeconds * 1000;
        document.querySelector('form[data-type="Video"]').submit();
        //document.querySelector('form').submit();
    } catch (error) {
        alert('Invalid video link');
    }
}

File: SWP391-SE1633NET-G2-main/LearningProject/src/java/Controller/AddLesson.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/JSP_Servlet/Servlet.java to edit this template
 */
package Controller;

import Model.Docs;
import Model.Lesson;
import Model.Quiz;
import Model.Section;
import Model.User;
import Model.Video;
import dal.DocsDAO;
import dal.LessonDAO;
import dal.QuizDAO;
import dal.SectionDAO;
import dal.VideoDAO;
import java.io.IOException;
import jakarta.servlet.ServletException;
import jakarta.servlet.http.HttpServlet;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import java.util.ArrayList;

/**
 *
 * @author Hieu Shin
 */
public class AddLesson extends HttpServlet {

    /**
     * Processes requests for both HTTP <code>GET</code> and <code>POST</code>
     * methods.
     *
     * @param request servlet request
     * @param response servlet response
     * @throws ServletException if a servlet-specific error occurs
     * @throws IOException if an I/O error occurs
     */
    protected void processRequest(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
    }

    // <editor-fold defaultstate="collapsed" desc="HttpServlet methods. Click on the + sign on the left to edit the code.">
    /**
     * Handles the HTTP <code>GET</code> method.
     *
     * @param request servlet request
     * @param response servlet response
     * @throws ServletException if a servlet-specific error occurs
     * @throws IOException if an I/O error occurs
     */
    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        if (request.getSession().getAttribute("user") != null) {
            User user = (User) request.getSession().getAttribute("user");
            if (user.getRole().equals("Admin")) {
                int sectionId = 0;
                int courseID = 0;
                if (request.getParameter("sectionID") != null && request.getParameter("courseID") != null) {
                    ArrayList<Lesson> lessons = new ArrayList<>();
                    ArrayList<Video> videolist = new ArrayList<Video>();
                    ArrayList<Docs> docslist = new ArrayList<Docs>();
                    ArrayList<Quiz> quizlist = new ArrayList<Quiz>();
                    
                    sectionId = Integer.parseInt(request.getParameter("sectionID"));
                    courseID = Integer.parseInt(request.getParameter("courseID"));
                    SectionDAO sectionDAO = new SectionDAO();
                    LessonDAO ldao = new LessonDAO();
                    VideoDAO vd = new VideoDAO();
                    DocsDAO dd = new DocsDAO();
                    QuizDAO qd = new QuizDAO();
                    lessons = ldao.getAllLessonOfSection(sectionId);
                    Section section = sectionDAO.getSectionBySectionID(sectionId);
                    
                    for (Lesson lesson : lessons) {
                        if (lesson.getType().equals("Video")) {
                            videolist.add(vd.getVideoOfLesson(lesson.getLessonId()));
                        } else if (lesson.getType().equals("Docs")) {
                            docslist.add(dd.getDocsOfLesson(lesson.getLessonId()));
                        } else {
                            quizlist.add(qd.getQuizOfLesson(lesson.getLessonId()));
                        }
                        
                    }
                    
                    request.setAttribute("quizlist", quizlist);
                    request.setAttribute("videolist", videolist);
                    request.setAttribute("docslist", docslist);
                    request.setAttribute("courseID", courseID);
                    request.setAttribute("lessons", lessons);
                    request.setAttribute("sectionID", sectionId);
                    request.setAttribute("section", section);
                    request.getRequestDispatcher("CreateLesson.jsp").forward(request, response);
                    return;
                }
                response.sendRedirect("AddLesson?courseID=" + courseID + "&sectionID=" + sectionId);
            } else {
                response.sendRedirect("home");
            }
        } else {
            response.sendRedirect("home");
        }
    }

    /**
     * Handles the HTTP <code>POST</code> method.
     *
     * @param request servlet request
     * @param response servlet response
     * @throws ServletException if a servlet-specific error occurs
     * @throws IOException if an I/O error occurs
     */
    @Override
    protected void doPost(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        int sectionId = Integer.parseInt(request.getParameter("sectionID"));
        int courseID = Integer.parseInt(request.getParameter("courseID"));
        String type = request.getParameter("type");
        
        LessonDAO ldao = new LessonDAO();
        
        if (type != null) {
            if (type.compareTo("Video") == 0) {
                String videotitle = request.getParameter("video_title");
                String videolink = request.getParameter("video_url");
                int duration = Integer.parseInt(request.getParameter("duration"));
                ldao.addLessonVideo(sectionId, videotitle, videotitle, videolink, duration);
            }
            
            if (type.compareTo("Docs") == 0) {
                String lesson_tilte = request.getParameter("lesson_tilte");
                int time_to_read = Integer.parseInt(request.getParameter("time_to_read"));
                String lesson_content = request.getParameter("lesson_content");
                ldao.addLessonDoc(sectionId, lesson_tilte, time_to_read, lesson_content);
            }
            if (type.compareTo("Quiz") == 0) {
                String lesson_title = request.getParameter("lessonTitle");
                int lesson_time = Integer.parseInt(request.getParameter("lessonTime"));
                ldao.addLessonQuiz(new Lesson(0, sectionId, lesson_title, false, "Quiz", false, lesson_time));
                int lessonID = ldao.getNewestLessonID(sectionId);
                QuizDAO quizDAO = new QuizDAO();
                quizDAO.addQuiz(lessonID);
            }
        }
        
        if (request.getParameter("button") != null) {
            if (request.getParameter("button").equals("Update document")) {
                int lessonId = Integer.parseInt(request.getParameter("LessonDocsId"));
                String lessonName = request.getParameter("LessonDocsName");
                int time = Integer.parseInt(request.getParameter("Time"));
                String docsContent = request.getParameter("DocsContent");
                
                ldao.updateLessonDocs(lessonName, time, docsContent, lessonId);
            }
            if (request.getParameter("button").equals("Update video")) {
                int lessonId = Integer.parseInt(request.getParameter("LessonVideoId"));
                String lessonName = request.getParameter("LessonVideoName");
                String videoLink = request.getParameter("videolink");
                int time = Integer.parseInt(request.getParameter("time_duration"));
                
                ldao.updateLessonVideo(lessonName, videoLink, lessonId, time);
                
            }
            if (request.getParameter("button").equals("Update quiz")) {
                int lessonId = Integer.parseInt(request.getParameter("LessonQuizId"));
                String lessonName = request.getParameter("LessonQuizTitle");
                int time = Integer.parseInt(request.getParameter("LessonQuizTime"));
                
                ldao.updateLessonQuiz(lessonName, time, lessonId);
                
            }
            
        }
        
        response.sendRedirect("AddLesson?courseID=" + courseID + "&sectionID=" + sectionId);
    }

    /**
     * Returns a short description of the servlet.
     *
     * @return a String containing servlet description
     */
    @Override
    public String getServletInfo() {
        return "Short description";
    }// </editor-fold>

}
